Model/AddressResolver.php

```diff
@@ -137,12 +137,12 @@ public function execute(DataObject $amwalOrderData, ?string $customerId = null):
 
     /**
      * @param DataObject $amwalOrderData
-     * @param null|string $customerId
+     * @param int|string|null $customerId
      * @return AddressInterface
      * @throws LocalizedException
      * @throws NoSuchEntityException
      */
-    public function createAddress(DataObject $amwalOrderData, ?string $customerId): AddressInterface
+    public function createAddress(DataObject $amwalOrderData, mixed $customerId = null): AddressInterface
```

Comment:
Mixed as a typehint has only been added in PHP 8.0 I believe, so for backwards compatibility we can drop the mixed type hint here and in other places
```suggestion
    public function createAddress(DataObject $amwalOrderData, $customerId = null): AddressInterface
```

Reply:
https://wiki.php.net/rfc/mixed-typehint, I think it support PHP 7